core/src/callback.jl

```diff
@@ -393,22 +398,131 @@ function set_fractional_flow_in_allocation!(
     return nothing
 end
 
-function set_control_params!(p::Parameters, node_id::NodeID, control_state::String)
-    node = getfield(p, p.graph[node_id].type)
-    idx = searchsortedfirst(node.node_id, node_id)
-    new_state = node.control_mapping[(node_id, control_state)]
+function update_parameters!(pump::Pump, parameter_update::ParameterUpdate)::Nothing
+    if parameter_update.node_type != NodeType.Pump
+        return nothing
+    end
+    (; node_idx, active, flow_rate_scalar) = parameter_update
+    pump.active[node_idx] = active
+    if !isnan(flow_rate_scalar)
```

Comment:
Why do we need to check for isnan? Not sure where it comes from, but ideally we don't even trigger when we have nan data?

Reply:
@SouthEndMusic Looking through the code, I don't think this can ever be nan? While it looks like it because of the ParameterUpdate struct, it is actually set for the correct nodetype using `parse_static_and_time`. In the AGV model any asserts didn't trigger on nan.